WikiWatershed/model-my-watershed
# Style observation popup and modal

This PR changes the style of the observations popups and plot modals according to wireframes in #1020, with the following exceptions. Note that only the final three commits were added by this PR, and the others are from the feature/observations branch.
* I omitted the sparklines in the popups because the data for each measurement takes a few seconds to load, and the scale and unit for each measurement is different which means the lines would be useless for comparison.
* I also omitted the last updated time on the popup because each measurement has a different time stamp (but implemented it for the plot modal since it only shows one measurement at a time). 
* I attempted to implement the hovering behavior for showing the depth of measurements, but there was a bug (only present in Chrome) which I could not fix after much effort, so I just put that info in a third column. I noticed that the depth is always 0.9m in the data sources we're using which is probably erroneous. Maybe we should not display this info?
* The CSS will be fine-tuned by @jfrankl in a separate commit later in the sprint to get the layout, fonts, and button icon to look good.

![screen shot 2015-11-25 at 10 43 50 am 2](https://cloud.githubusercontent.com/assets/1896461/11402790/a30ffd72-9367-11e5-9b8c-cd65c614bc4c.png)

![screen shot 2015-11-25 at 10 44 01 am 2](https://cloud.githubusercontent.com/assets/1896461/11402787/9f401042-9367-11e5-9032-df50f1292dd8.png)

![screen shot 2015-11-25 at 11 00 22 am 2](https://cloud.githubusercontent.com/assets/1896461/11402791/a604e8ee-9367-11e5-8f21-3ee4bbd812b5.png)


##### Testing
* Click on an observation marker.
* Click on a row which should open the plot modal with that measurement selected.
* A spinner should appear while loading.
* Selecting from the dropdown should load another plot. Selecting a new measurement while loading is in progress should work.
* Selecting a previously displayed measurement from the dropdown should not require loading since the data is cached.
* Closing the modal and reopening should work as expected.
* Clicking the 'Historical Data' button should open the plot modal with the first element in the dropdown selected.

Connects #1020
